component/src/main/java/io/siddhi/extension/io/file/listeners/FileSystemListener.java

```diff
@@ -227,31 +228,32 @@ public void run() {
     private void reProcessFile(VFSClientConnector vfsClientConnector,
                                VFSClientConnectorCallback vfsClientConnectorCallback,
                                Map<String, String> properties, String fileUri) {
-        properties.put(Constants.URI, fileUri);
-        properties.put(Constants.ACK_TIME_OUT, "1000");
-        BinaryCarbonMessage carbonMessage = new BinaryCarbonMessage(ByteBuffer.wrap(
-                fileUri.getBytes(StandardCharsets.UTF_8)), true);
-
         String actionAfterProcess = fileSourceConfiguration.getActionAfterProcess();
-        String moveAfterProcess = fileSourceConfiguration.getMoveAfterProcess();
-        try {
-            if (fileSourceConfiguration.getActionAfterProcess() != null) {
-                properties.put(Constants.URI, fileUri);
-                properties.put(Constants.ACTION, actionAfterProcess);
-                if (fileSourceConfiguration.getMoveAfterProcess() != null) {
-                    String destination = constructPath(moveAfterProcess, getFileName(fileUri, fileSourceConfiguration
-                            .getProtocolForMoveAfterProcess()));
-                    if (destination != null) {
-                        properties.put(Constants.DESTINATION, destination);
+        if (!actionAfterProcess.equalsIgnoreCase(Constants.KEEP)) {
```

Comment:
Can we move this if to the place where we invoke reProcess method

Reply:
@tishan89 this is fixed with https://github.com/siddhi-io/siddhi-io-file/pull/81/commits/06a8e61c30d18257b68c4531868f80528d83a026